### ISSUE TITLE
API to sort alignment tracks by base

### ISSUE THREAD

**Issue by ggydush** [igvteam/igv.js#893]
Sorry, I may be missing something, but I am trying to sort all alignment tracks by base for varying positions. I jump to each position using: `browser.search(site)`

Is it there an API to sort alignment tracks by base? I know the UI allows this, but it great if we could sort after each search automatically.

**Comment by jrobinso** [comment]
@ggydush This should be there, but isn't.   I was surprised myself!   I will put it on the high priority list.

**Comment by jrobinso** [comment]
bump

**Comment by jrobinso** [comment]
@ggydush Sorry for the long delay!  I've started working on this in a branch, its a little complicated due to the asynchronous nature of searching and alignment loading.

**Comment by brentp** [comment]
This would be quite useful! Or to have a configuration option to sort by base each time a new locus is set.

**Comment by Christian-Sidak** [comment]
Has this been added? This issue remains open, but it appears from this wiki page that sorting by base should be possible: https://github.com/igvteam/igv.js/wiki/Alignment-Track 

Provided your option block is formatted something like: 

{
  "name": “Sample 1”,
  "showNavigation": true,
  "showRuler": true,
  "genome": "hg19",
  "locus": "chr1:13718415",
  "tracks": [
    {
      "url": “sample1.bam",
      "indexURL": “sample1.bai",
      "type": "alignment",
      "format": "bam",
      "name": “sample1.bam",
      "displayMode": "SQUISHED",
      "alignmentRowHeight": 4,
      "sort": {
        "locus": "chr1:13718415",
        "option": "NUCLEOTIDE",
        "dirction": "DESC"
      }
    }

However, adding a sort option does not seem to do anything to the tracks.

**Comment by jrobinso** [comment]
@Christian-Sidak first sorry I missed your comment.   Adding a sort option has been in place and working for some time, if its not working for you I will need a test case.     This issue is for an API to sort dynamically, at least that's how I interpret it.   That is also implemented in the master branch and will be released soon.

**Comment by jrobinso** [comment]
This has been implemented with a "sort" function on Browser and will be released soon.   Example usage below,  not the sort object properties have changed (explicit chr and position instead of locus string).   However the old syntax will still work

```
<button id="gotoButton">Goto chr1:155,155,358</button>
<button id="button1">Sort position chr1:155,155,358</button>
<button id="button2">Sort position chr1:155,155,361</button>
<div id="igvDiv" style="padding-top: 50px;padding-bottom: 20px; height: auto"></div>

...


   var options =
        {
            genome: "hg19",
             locus: [
                "chr1:155,155,322-155,155,392"
            ],
            tracks: [
                {
                    id: "myBamTrack",
                    type: "alignment",
                    url: "gs://genomics-public-data/platinum-genomes/bam/NA12878_S1.bam",
                    indexURL: "gs://genomics-public-data/platinum-genomes/bam/NA12878_S1.bam.bai",
                    name: "NA12878",
                    colorBy: "strand",
                    // Uncomment to set initial sort
                    // sort: [
                    //     {
                    //         chr: "chr1",
                    //         position: 155155358,
                    //         option: "BASE",
                    //         direction: "ASC"
                    //     }
                    // ]
                }
            ]
        };

    var igvDiv = document.getElementById("igvDiv");

    igv.createBrowser(igvDiv, options)
        .then(function (browser) {
            console.log("Created IGV browser");

            document.getElementById("gotoButton").addEventListener("click", () => browser.search("chr1:155,155,322-155,155,392"));

            document.getElementById("button1").addEventListener("click", () => {
                const bamTrack = browser.findTracks("id", "myBamTrack")[0];
                bamTrack.sort({
                chr: "chr1",
                position: 155155358,
                option: "BASE",
                direction: "ASC"
            });})

            document.getElementById("button2").addEventListener("click", () => {
                const bamTrack = browser.findTracks("id", "myBamTrack")[0];
                bamTrack.sort({
                chr: "chr1",
                position: 155155361,
                option: "BASE",
                direction: "ASC"
            });})

        })

**Comment by jrobinso** [comment]
This is done and available in the master branch.  It will be released in 2.7.0.
